catalyst-gateway/bin/src/db/event/config/key.rs

```diff
@@ -43,28 +45,31 @@ static FRONTEND_IP_DEFAULT: LazyLock<Value> =
 /// Helper function to create a JSON validator from a JSON schema.
 /// If the schema is invalid, a default JSON validator is created.
 fn schema_validator(schema: &Value) -> Validator {
-    jsonschema::validator_for(schema).unwrap_or_else(|err| {
-        error!(
-            id = "schema_validator",
-            error=?err,
-            "Error creating JSON validator"
-        );
-
-        // Create a default JSON validator as a fallback
-        // This should not fail since it is hard coded
-        #[allow(clippy::expect_used)]
-        Validator::new(&json!({
-            "$schema": "http://json-schema.org/draft-07/schema#",
-            "type": "object"
-        }))
-        .expect("Failed to create default JSON validator")
-    })
+    Validator::options()
+        .with_draft(jsonschema::Draft::Draft202012)
+        .build(schema)
+        .unwrap_or_else(|err| {
+            error!(
+                id="schema_validator",
+                error=?err,
+                "Error creating JSON validator"
+            );
+
+            // Create a default JSON validator as a fallback
+            // This should not fail since it is hard coded
+            #[allow(clippy::expect_used)]
+            Validator::new(&json!({
```

Comment:
Seems it make sense to move this code under its own function like and add a unit test for this function.
So then we will 100% sure that it will never fail.
```Rust
 #[allow(clippy::expect_used)]
fn default_json_validator() -> Validator {
 Validator::new(&json!({  "$schema": SCHEMA_VERSION,
                "type": "object"
            }))
            .expect("Failed to create default JSON validator")
}
```